### PR TITLE
Update CONTRIBUTING.md

### LINKED ISSUE #2663: Update CONTRIBUTING.md
we should improve this doc:
[contributing.md](https://github.com/marionettejs/backbone.marionette/blob/master/CONTRIBUTING.md)

#### should have a table of contents
we've got lots of material but it's hard to follow. it'd be better if there were a TOC on the top so that we could see all the material.

#### should have the steps for getting started:

```
1. go to the project
2. git clone the repo
3. install libraries
4. run tests
5. create a branch
6. make a change
7. run tests
8. create a commit
9. fork marionette
10. push branch to remote
11. make a pull request on project
```

the eslint people have some good content too:
http://eslint.org/docs/developer-guide/contributing.html#pull-requests

### PR DESCRIPTION
**Pull request by cmisenas**
Resolves #2663 

I don't think this is the final yet but wanted to get some feedback about it. So far, I only added a table of contents, rearranged some content and tried correcting some grammar/typos from old CONTRIBUTING.md. 

Particularly want to hear feedback about:
* format/contents of doc
* is there a convention/style guide?
* what major branch is
* should it also state one commit per PR?
* is talking about issues/prs on gitter encouraged?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,151 +1,132 @@
...
-terrible with), just click the "Edit this file" button
-and send in the pull request for the fix. For larger
-changes and big swaths of documentation changes, a regular
-pull request as outlined below is more appropriate.
+* Marionette version no.
+* Backbone version no.
+* jQuery version no.
+* Underscore version no.
+* Browser and version no.
+* Operating System and version no.
```
Comment (jasonLaster) [inline CONTRIBUTING.md]: The version requirements seem pretty dated. I'd say, we only care about Mn and BB. 90% of the time. I'd like to see the others dropped because they add a level of intimidation (oh no, maybe my bug happened bc I'm running leopard)

**Comment by jasonLaster** [comment]
Thanks Jade! 

I really like these changes. The TOC really helps and the re-arranging you did made it much clearer. 

Legit :+1:

**Comment by cmisenas** [comment]
:+1: @jasonLaster 
Changed version requirements.

**Comment by jasonLaster** [comment]
I don't want to introduce scope creep, but I just made a little dude that could go on the top of the page

```md
<img style="position:center" src="https://www.dropbox.com/s/wj68so87r1aofdh/marionette%20construction%20worker.jpg?dl=0" width="128" height="128">
```

**Comment by samccone** [comment]
:+1: land

**Comment by samccone** [comment]
thanks @cmisenas !!!!!!!!!

**Comment by cmisenas** [comment]
:dancers:

**Comment by jdaudier** [comment]
Thanks @cmisenas!  Appreciate the help.

**Comment by cmisenas** [comment]
Happy to help. Also, cool little marionette dude @jasonLaster :D

**Comment by jdaudier** [comment]
@jasonLaster Did you want to have Jade add that to the contributing doc?

**Comment by jasonLaster** [comment]
@cmisenas @jdaudier - yea, i think that'd be nice :)

I added a comment for that and an additional photo.
https://github.com/marionettejs/backbone.marionette/issues/2663#issuecomment-127655551
